bundles/binding/org.openhab.binding.powermax/src/main/java/org/openhab/binding/powermax/internal/PowerMaxBinding.java

```diff
@@ -0,0 +1,907 @@
+                                }
+                                break;
+                            case PGM_STATUS:
+                                if ((item instanceof SwitchItem) && (state.getPGMX10DeviceStatus(0) != null)) {
+                                    itemState = state.getPGMX10DeviceStatus(0) ? OnOffType.ON : OnOffType.OFF;
+                                }
+                                break;
+                            case X10_STATUS:
+                                if ((num != null) && (state.getPGMX10DeviceStatus(num) != null)) {
+                                    if (item instanceof SwitchItem) {
+                                        itemState = state.getPGMX10DeviceStatus(num) ? OnOffType.ON : OnOffType.OFF;
+                                    } else if (item instanceof StringItem) {
+                                        itemState = new StringType(state.getPGMX10DeviceStatus(num) ? "ON" : "OFF");
+                                    }
+                                }
+                                break;
+                            default:
+                                break;
+                        }
+
+                        if (itemState != null) {
+                            eventPublisher.postUpdate(itemName, itemState);
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Post item updates on the bus to match the alarm panel settings
+     */
+    private void updateItemsFromPanelSettings() {
+        updateItemsFromPanelSettings(null, null, null);
+    }
+
+    /**
+     * Post item updates on the bus to match the alarm panel settings
+     *
+     * @param provider
+     *            filter on provider; if null, no filter on provider
+     * @param name
+     *            filter on item name; if null, no filter on item name
+     * @param selector
+     *            filter on selector type; if null, no filter on selector type
+     */
+    private synchronized void updateItemsFromPanelSettings(PowerMaxBindingProvider provider, String name,
+            PowerMaxSelectorType selector) {
+
+        if (itemUpdateDisabled) {
+            logger.debug("updateItemsFromPanelSettings(): items update disabled");
+            return;
+        }
+
+        if (provider == null) {
+            for (PowerMaxBindingProvider prov : providers) {
+                updateItemsFromPanelSettings(prov, name, selector);
+            }
+        } else {
+            PowerMaxPanelSettings settings = PowerMaxPanelSettings.getThePanelSettings();
+
+            for (String itemName : provider.getItemNames()) {
+                if ((name == null) || itemName.equals(name)) {
+                    State itemState = null;
+
+                    PowerMaxBindingConfig config = provider.getConfig(itemName);
+                    Item item = config.getItem();
```

Comment:
Another case where TypeParser could be used and you could support any kind of item by just using its list of accepted data types.